### ISSUE TITLE
Climate entities no longer support set target temperature

### ISSUE THREAD

**Issue by mrcushing** [MizterB/homeassistant-infinitude-beyond#25]
Ever since the latest few updates (updating infinitude server + latest beyond updates), the climate entities no longer support the service "set target temperature" and kick off an error.  

I had a script that would set the target temperature of, e.g., the bedroom thermostat to 76.  It was set up in the UI.  I've recently looked at the script, and the target temperature option has disappeared from the UI.  In YAML, the target temperature is still there from before.  When I run the script (or any others that were written before the updates that set target temperature), it kicks off an error saying that the climate entities do not support setting the target temperature.

When I go to developer tools, pull up the action tab, and go to "climate: set target temperature," none of the thermostats show up as possible entities to test that action on.  Other non-infinitude climate entities still show the set temperature climate action as available.

Log attached.
[home-assistant_infinitude_beyond_2024-08-28T12-22-38.618Z.log](https://github.com/user-attachments/files/16783272/home-assistant_infinitude_beyond_2024-08-28T12-22-38.618Z.log)

Edit:  HA 2024.8.3 - I wonder if it happened with the recent HA switch in terminology from call service to actions, and whether that is causing the issue?

**Comment by mrcushing** [comment]
YAML of the script, if it matters:

alias: HVAC - reduce energy use (76 and 66)
sequence:
  - if:
      - condition: state
        entity_id: climate.master_bedroom_thermostat
        state: cool
      - condition: state
        entity_id: climate.living_room_thermostat
        state: cool
    then:
      - target:
          entity_id:
            - climate.living_room_thermostat
        data:
          temperature: 76
        action: climate.set_temperature
      - metadata: {}
        data:
          temperature: 76
        target:
          entity_id:
            - climate.basement_thermostat
            - climate.master_bedroom_thermostat
        action: climate.set_temperature
    else:
      - target:
          entity_id:
            - climate.living_room_thermostat
        data:
          temperature: 66
        action: climate.set_temperature
      - metadata: {}
        data:
          temperature: 66
        target:
          entity_id:
            - climate.basement_thermostat
            - climate.master_bedroom_thermostat
        action: climate.set_temperature
  - target:
      entity_id:
        - climate.basement_thermostat
        - climate.master_bedroom_thermostat
        - climate.living_room_thermostat
    data:
      preset_mode: Hold indefinitely
    action: climate.set_preset_mode
description: Reduces HVAC use by raising cool to 76 and lowering heat to 66.

**Comment by MizterB** [comment]
What is your thermostat mode - Auto/Heat/Cool?

**Comment by mrcushing** [comment]
Cool, at the moment that script runs (I never run auto, only heat or cool with a separate automation that switches between them)

**Comment by MizterB** [comment]
Hmmm....still trying to make sense of this.  Just did a full slate of tests here... No issues setting target temperature for Auto/Cool/Heat.  However, if Off or Fan Only, the service call will fail.

**Comment by mrcushing** [comment]
Why would the climate entity (infinitude beyond) not report to HA that it is capable of setting temperature? Is there a way to "tell" HA that the entity is capable of the action set temperature?

**Comment by MizterB** [comment]
Unsure what the trick is...  I've also discovered that none of my climate integrations (I have Infinitude Beyond, Kumo, and Google Nest) are providing climate entities to the Actions UI under Development Tools.

But you indicated that you have other climate entities that are working correctly?  Which integrations?

**Comment by MizterB** [comment]
> Why would the climate entity (infinitude beyond) not report to HA that it is capable of setting temperature? Is there a way to "tell" HA that the entity is capable of the action set temperature?

There is a "Supported Features" attribute that provides these values - I may be missing something.

**Comment by mrcushing** [comment]
Tesla provides a climate entity that works with the actions UI

**Comment by MizterB** [comment]
Is that a custom integration?  If so, can you please provide a link to it?

**Comment by mrcushing** [comment]
The odd thing is, if I press "+" on the climate entity card, it will in fact change the set temperature and hold that temperature on my system.  So HA can send a "increase temperature" command, at least - even if it thinks it cannot send a "set temperature to X" action.

**Comment by mrcushing** [comment]
> Is that a custom integration? If so, can you please provide a link to it?

https://github.com/alandtse/tesla

**Comment by MizterB** [comment]
Well, like I mentioned before, if using YAML, I am able to perform the set_temperature action just fine, as long as the mode is correct.  Something may be unique about your setup...not sure yet

**Comment by mrcushing** [comment]
> The odd thing is, if I press "+" on the climate entity card, it will in fact change the set temperature and hold that temperature on my system. So HA can send a "increase temperature" command, at least - even if it thinks it cannot send a "set temperature to X" action.

Unbelievably, now that I have manually increased the temperature on the climate card, HA suddenly "sees" infinitude beyond as capable of setting a target temperature again.  I have no idea why.  Testing my script again to see if it works.

**Comment by mrcushing** [comment]
Yup, everything works again.  So, if someone has a problem with a set target temperature action not appearing through the developer tools section, just try manually triggering an increase in temperature through the climate card and apparently HA will learn that action is available and make it appear in the developer tools actions UI section again.

**Comment by MizterB** [comment]
Unable to reproduce that behavior over here...still don't have devices showing up in the UI.  Will continue to investigate.
